numpy/lib/_iotools.py

```diff
@@ -519,11 +519,16 @@ class StringConverter(object):
     #
     _mapper = [(nx.bool_, str2bool, False),
                (nx.integer, int, -1),
+               (nx.int64, int, -1),
```

Comment:
I think you should only include the `np.int64` entry if it is going to be different from `nx.integer`, e.g. build the `_mapper` list up to and including the `nx.integer` entry, then do something like:

```
if np.dtype(nx.integer).itemsize < np.dtype(nx.int64):
    _mapper.append((nx.int64, int, -1))
```

then extend the list with the remaining items, from the `nx.uint64` entry to the end.